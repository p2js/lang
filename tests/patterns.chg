type Unnested = enum { A, B, C };

fn main(): i32 {
    let x = Unnested::A;
    let h = match x {
        Unnested::A => 1,
        Unnested::B => 2,
        other => 4
    };
    return h;
}
