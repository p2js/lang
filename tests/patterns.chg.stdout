expr @ 4:23 4:24 => Unnested
expr @ 5:11 5:12 => Unnested
expr @ 6:31 6:32 => i32
expr @ 6:24 6:32 => never
expr @ 7:33 7:34 => i32
expr @ 7:26 7:34 => never
expr @ 7:24 8:0 => ()
expr @ 8:25 9:0 => i32
expr @ 8:18 9:0 => never
expr @ 5:5 9:6 => ()
expr @ 3:16 11:1 => ()



define i32 @main() {
prologue:
%l.0 = alloca i32
%l.1 = alloca i8
%l.6 = alloca i8
br label %bb.0

bb.0:
store i8 0, i8* %l.1
br label %bb.1

bb.1:
%l.2 = load i8, i8* %l.1
%l.3 = icmp eq i8 %l.2, 0
br i1 %l.3, label %bb.4, label %bb.2

bb.2:
%l.4 = load i8, i8* %l.1
%l.5 = icmp eq i8 %l.4, 1
br i1 %l.5, label %bb.5, label %bb.3

bb.3:
br label %bb.6

bb.4:
store i32 1, i32* %l.0
%ret.0 = load i32, i32* %l.0
ret i32 %ret.0

bb.5:
store i32 2, i32* %l.0
%ret.1 = load i32, i32* %l.0
ret i32 %ret.1

bb.6:
%l.7 = load i8, i8* %l.1
store i8 %l.7, i8* %l.6
store i32 4, i32* %l.0
%ret.2 = load i32, i32* %l.0
ret i32 %ret.2

bb.8:
br label %bb.7

bb.9:
br label %bb.7

bb.10:
br label %bb.7

}


